lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,21 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
```

Comment:
Gotcha. I can implement it as a blacklist of characters not allowed as the first and/or second character (with slightly different requirements for first and second). However, I'm not sure where I'd find a complete list of such characters.